src/proto/h1/role.rs

```diff
@@ -359,16 +384,51 @@ impl Http1Transaction for Server {
         let init_cap = 30 + msg.head.headers.len() * AVERAGE_HEADER_SIZE;
         dst.reserve(init_cap);
         if msg.head.version == Version::HTTP_11 && msg.head.subject == StatusCode::OK {
+            #[cfg(feature = "traceable")]
+            span.as_ref().unwrap().record("http.flavor", &1.1);
+
             extend(dst, b"HTTP/1.1 200 OK\r\n");
         } else {
             match msg.head.version {
-                Version::HTTP_10 => extend(dst, b"HTTP/1.0 "),
-                Version::HTTP_11 => extend(dst, b"HTTP/1.1 "),
+                Version::HTTP_10 => {
+                    #[cfg(feature = "traceable")]
+                    span.as_ref().unwrap().record("http.flavor", &1.0);
+
+                    extend(dst, b"HTTP/1.0 ");
+                }
+                Version::HTTP_11 => {
+                    #[cfg(feature = "traceable")]
+                    span.as_ref().unwrap().record("http.flavor", &1.1);
+
+                    extend(dst, b"HTTP/1.1 ");
+                }
                 Version::HTTP_2 => {
-                    debug!("response with HTTP2 version coerced to HTTP/1.1");
+                    #[cfg(feature = "traceable")]
+                    {
+                        span.as_ref().unwrap().record("http.flavor", &1.1);
+                        event!(
+                            tracing::Level::TRACE,
+                            "response with HTTP2 version coerced to HTTP/1.1"
+                        );
+                    }
                     extend(dst, b"HTTP/1.1 ");
                 }
-                other => panic!("unexpected response version: {:?}", other),
+                other => {
+                    #[cfg(feature = "traceable")]
+                    {
+                        // 0.0 is a placeholder for `other` until `impl Debug for Version`
+                        span.as_ref().unwrap().record("http.flavor", &0.0);
+                        span.as_ref().unwrap().record(
+                            "otel.status_code",
+                            &format!("{:?}", opentelemetry::trace::StatusCode::Error).as_str(),
+                        );
+                        span.as_ref().unwrap().record(
+                            "otel.status_message",
+                            &format!("PANIC: unexpected response version: {:?}", other).as_str(),
+                        );
```

Comment:
Note to self (@taqtiqa-mark): Use this error to simultaneously try out the ergonomics of the alternative syntax for traceable errors.

Come to think of it - is there any reason not to provide both and let users adopt as they prefer? 
The implementation details should be shareable.